ast/src/parsed/display.rs

```diff
@@ -539,6 +539,52 @@ fn format_list<L: IntoIterator<Item = I>, I: Display>(list: L) -> String {
     format!("{}", list.into_iter().format(", "))
 }
 
+impl<E: Display> Expression<E> {
+    pub fn binary_op_precedence(&self) -> ExpressionPrecedence {
```

Comment:
I think it might be better to add this to the impl for BinaryOperator - and you can put it directly in the mod.rs, it's not just related to Display.